tools/export/iar/__init__.py

```diff
@@ -106,14 +105,20 @@ def generate(self):
         #Optimizations
         if '-Oh' in flags['c_flags']:
             flags['c_flags'].remove('-Oh')
+
+        try:
+            debugger = DeviceCMSIS(self.target).debug.replace('-','').upper()
+        except:
```

Comment:
Perhaps. I'd kind of rather that this bit of code never fails. You should always get a debug interface, at least CMSIS-DAP. But I could catch the existing thrown exception, `TargetNotSupported`.

https://github.com/ARMmbed/mbed-os/blob/master/tools/export/cmsis/__init__.py#L38